generator/lkml_update.py

```diff
@@ -0,0 +1,49 @@
+"""An updated lkml parser to handle explore queries."""
+from typing import List, Union
+
+from lkml.keys import KEYS_WITH_NAME_FIELDS
+from lkml.simple import DictParser
+from lkml.tree import BlockNode, DocumentNode, ListNode, PairNode
+
+
+def dump(obj: dict) -> str:
+    """Dump an object as LookML."""
+    parser = UpdatedDictParser()
+    tree: DocumentNode = parser.parse(obj)
+    return str(tree)
+
+
+class UpdatedDictParser(DictParser):
+    """An updated DictParser that properly handles queries."""
+
+    def parse_any(
+        self, key: str, value: Union[str, list, tuple, dict]
+    ) -> Union[
+        List[Union[BlockNode, ListNode, PairNode]], BlockNode, ListNode, PairNode
+    ]:
+        """Dynamically serializes a Python object based on its type.
+
+        Args:
+            key: A LookML field type (e.g. "suggestions" or "hidden")
+            value: A string, tuple, or list to serialize
+        Raises:
+            TypeError: If input value is not of a valid type
+        Returns:
+            A generator of serialized string chunks
+        """
+        if isinstance(value, str):
+            return self.parse_pair(key, value)
+        elif isinstance(value, (list, tuple)):
+            if self.is_plural_key(key) and not self.parent_key == "query":
```

Comment:
This line is the only one I changed, where I prevent it from expanding the list if the parent is a query. Issue is filed upstream.

Reply:
Can you add a reference to the issue you filed in a comment, both for this method and in the body of the class? That way someone in the future can reason about why this exists without digging into history.